cpp/include/raft/comms/test.hpp

```diff
@@ -246,5 +249,33 @@ bool test_pointToPoint_simple_send_recv(const handle_t &h, int numTrials) {
 
   return ret;
 }
+
+/**
+ * A simple test that the comms can be split into 2 separate subcommunicators
+ *
+ * @param the raft handle to use. This is expected to already have an
+ *        initialized comms instance.
+ */
+bool test_commsplit(const handle_t &h, int n_colors) {
+  const comms_t &communicator = h.get_comms();
+  const int rank = communicator.get_rank();
+  const int size = communicator.get_size();
+
+  if (n_colors > size) n_colors = size;
```

Comment:
This is just a test function to verify that the implementation is working. It doesn't really hurt anything if n_colors > size. It could be truncated here or in the Python layer but I thought it made sense to do it here.